src/pkgcheck/checks/profiles.py

```diff
@@ -411,9 +411,8 @@ def finish(self):
                 continue
             for parent in profile.stack:
                 seen_profile_dirs.update(dir_parents(parent.name))
-                # flag lagging profile EAPIs -- assumes EAPIs are sequentially
-                # numbered which should be the case for the gentoo repo
-                if (self.options.gentoo_repo and str(profile.eapi) < str(parent.eapi)):
+                if (str(profile.eapi) != str(parent.eapi) and
```

Comment:
Why is string coercing needed here? It should be possible to compare EAPI objects directly for this whole change.

Reply:
I didn't trust the code ;-).

Reply:
Ok, changed now.